packages/core/src/new-api/internal/validation/validate.ts

```diff
@@ -30,41 +30,77 @@ export async function validate(
   // validating each of those, we achieve the same effect.
   const submodules = module.submodules;
   for (const submodule of submodules) {
-    await validate(submodule, artifactLoader, moduleParameters);
+    await validate(submodule, artifactLoader, deploymentParameters);
   }
 
-  const params = moduleParameters[module.id] ?? {};
-
   for (const future of futures) {
     switch (future.type) {
       case FutureType.ARTIFACT_CONTRACT_DEPLOYMENT:
-        return validateArtifactContractDeployment(
+        await validateArtifactContractDeployment(
```

Comment:
What drove the switch here to using an `await` rather than returning?

Reply:
I think it's because they return `Promise<void>`

Reply:
This was the bug causing us to only validate a single future from each deployment. The reason being that we are looping through the futures so returning ends the loop early.